themes/classic/templates/catalog/_partials/product-details.tpl

```diff
@@ -3,7 +3,17 @@
     {if $product.reference}
       <div class="product-reference">
         <label class="label">{l s='Reference' d='Shop.Theme.Catalog'} </label>
-        <span itemprop="sku">{$product.reference}</span>
+        <span itemprop="sku">
+          {foreach from=$product.attributes item=att name=reference}
```

Comment:
Are you sure that it would work properly even if product has no attributes at all?